File: Add Code Here/LeetCode_Sols/Easy/144. Binary Tree Preorder Traversal/Solution.java
class Solution {
    List<Integer> order = new ArrayList<Integer>();
    public List<Integer> preorderTraversal(TreeNode root) {
        traverse(root);
        return order;
    }
    
    public void traverse(TreeNode root){
        if (root == null) return;
        order.add(root.val);
        traverse(root.left);
        traverse(root.right);
    }
}

File: Add Code Here/LeetCode_Sols/Easy/144. Binary Tree Preorder Traversal/Solution.kt
/**
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */
class Solution {
    fun preorderTraversal(root: TreeNode?): List<Int> {
        val result = mutableListOf<Int>()
        helper(root, result)
        return result
    }
    
    fun helper(node: TreeNode?, result: MutableList<Int>) {
        if (node == null) return
        result.add(node.`val`)
        helper(node.left, result)
        helper(node.right, result)
    }
}